SonicWall/sonicwall-fw/ingest/parser.yml

```
@@ -0,0 +1,355 @@
+          90: "sprite-rpc"
+          91: "larp"
+          92: "mtp"
+          93: "ax.25"
+          94: "ipip"
+          95: "micp"
+          96: "scc-sp"
+          97: "etherip"
+          98: "encap"
+          99: "any private encryption scheme"
+          100: "gmtp"
+          101: "ifmp"
+          102: "pnni"
+          103: "pim"
+          104: "aris"
+          105: "scps"
+          106: "qnx"
+          107: "a/n"
+          108: "ipcomp"
+          109: "snp"
+          110: "compaq-peer"
+          111: "ipx-in-ip"
+          112: "vrrp"
+          113: "pgm"
+          114: "any 0-hop protocol"
+          115: "l2tp"
+          116: "ddx"
+          117: "iatp"
+          118: "stp"
+          119: "srp"
+          120: "uti"
+          121: "smp"
+          122: "sm"
+          123: "ptp"
+          124: "isis over ipv4"
+          125: "fire"
+          126: "crtp"
+          127: "crudp"
+          128: "sscopmce"
+          129: "iplt"
+          130: "sps"
+          131: "pipe"
+          132: "sctp"
+          133: "fc"
+          134: "rsvp-e2e-ignore"
+          135: "mobility header"
+          136: "udplite"
+          137: "mpls-in-ip"
+          138: "manet"
+          139: "hip"
+          140: "shim6"
+          141: "wesp"
+          142: "rohc"
+          253: "unassigned"
+          254: "use for experimentation and testing"
+          255: "reserved"
+        mapping:
+          parsed_event.message.proto: network.transport
+
+      - translate:
+        dictionary:
+          0: 'base event'
+          1: 'aggregated'
+          2: 'correlation'
+          3: 'action'
+        mapping:
+          parsed_event.message.type: cef.event_type
+        fallback: 'base event'
+
+  set_cef_extension_fields_cef:
+    actions:
+      - set:
+          cef: '{{parsed_event.message}}'
+
+  set_ecs_fields:
+    actions:
+      - set:
+          destination.bytes: '{{parsed_event.message.in}}'
+          destination.domain: '{{parsed_event.message.request}}'
+          destination.ip: '{{parsed_event.message.dst or parsed_event.message.dstV6}}'
+          destination.mac: '{{parsed_event.message.dmac}}'
+          destination.nat.ip: '{{parsed_event.message.cs2Label or parsed_event.message.natDstV6}}'
+          destination.nat.port: '{{parsed_event.message.dnpt}}'
+          destination.packets: '{{parsed_event.message.cn1Label}}'
+          destination.port: '{{parsed_event.message.dpt}}'
+          network.transport: '{{parsed_event.message.proto}}'
+          observer.egress.interface.name: '{{parsed_event.message.deviceOutboundInterface}}'
+          observer.ingress.interface.name: '{{parsed_event.message.deviceInboundInterface}}'
+          observer.type: 'firewall'
+          # process.id: '{{parsed_event.message.app}}'
+          process.entity_id: '{{parsed_event.message.appid}}'
+          process.name: '{{parsed_event.message.appName}}'
+          rule.id: '{{parsed_event.message.cs1}}'
```

According to the tests, [`rule.name`](https://www.elastic.co/guide/en/ecs/8.4/ecs-rule.html#field-rule-name) seems more suitable.
```suggestion
          rule.name: '{{parsed_event.message.cs1}}'
```